### PR TITLE
Cleanup some GenericContainer warnings

### PR DESCRIPTION
**Pull request by deejgregor**
### Changes

This just cleans up some warnings I came across in our container classes that extend GenericContainers. It seemed easy to fix (which was *almost* true), so I cleaned them up.

The [MockCloudContainer](https://github.com/OpenNMS/opennms/pull/5678/commits/49df9f862bae8277ad15445367a0c63f1f769918) change wasn't something I expected to do, but it let me eliminate the last case of a raw type of GenericContainer being used. And overall, I like the change (I hope others agree, :grin:).

While moving MockCloudContainer into its own class, I noticed some duplication with installing KAR features, which led me to [Move installFeature into KarafContainer.](https://github.com/OpenNMS/opennms/pull/5678/commits/c2ae9e962d5f645411d013694033910333247b35) 

I also deduplicated the Jaeger Thrift HTTP while I was waiting for tests to run. :)

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/${JIRA-ISSUE-NUMBER}

### CODE REVIEW

**Comment by deejgregor** [comment]
@FrancescoBuc86 I wanted to see if you would mind reviewing mostly this commit: [Split mock cloud server into MockCloudContainer](https://github.com/OpenNMS/opennms/pull/5678/commits/65739678b12acece7e29dafa99e821d30cff01d0)

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+public class MockCloudContainer extends org.testcontainers.containers.GenericContainer<MockCloudContainer> {
```
Comment (fooker) [inline smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java:43]: Please import `GenericContainer` instead.

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                        + "property-set pas.mtls.host %s%n"
+                        + "property-set pas.mtls.port %s%n"
+                        + "property-set pas.mtls.security MTLS%n"
+                        + "property-set grpc.truststore \"%s\"%n"
+                        + "config:update",
+                ALIAS,
+                PORT,
+                ALIAS,
+                PORT,
+                classpathFileToString("/cert/horizon/servertruststore.pem")
```
Comment (fooker) [inline smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java:80]:
```suggestion
        return Joiner.on('\n').join(new String[] {
                "config:edit org.opennms.plugins.cloud",
                String.format("property-set pas.tls.host %s", ALIAS),
                String.format("property-set pas.tls.port %s", PORT),
                "property-set pas.tls.security TLS",
                String.format("property-set pas.mtls.host %s", ALIAS),
                String.format("property-set pas.mtls.port %s", PORT),
                "property-set pas.mtls.security MTLS",
                String.format("property-set grpc.truststore \"%s\"", classpathFileToString("/cert/horizon/servertruststore.pem")),
                "config:update",
        });
```

This moves the replacement into the line needed.

**Review by fooker — CHANGES_REQUESTED** [review]

File: smoke-test/src/main/java/org/opennms/smoketest/containers/KarafContainer.java [hunk 3]
```diff
@@ -28,12 +28,17 @@
...
 import java.net.InetSocketAddress;
+import java.nio.file.Path;
 
+import org.opennms.smoketest.utils.KarafShell;
 import org.opennms.smoketest.utils.SshClient;
 import org.testcontainers.containers.Container;
+import org.testcontainers.utility.MountableFile;
 
-public interface KarafContainer extends Container {
+public interface KarafContainer<T extends Container<T>> extends Container<T> {
```
[the remark this answers is not in the document]
Comment (deejgregor) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/KarafContainer.java]: Done.

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java (hunk 1 above)
Comment (deejgregor) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java]: Good catch. How did I miss this myself? :)

File: smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java (hunk 2 above)
Comment (deejgregor) [reply smoke-test/src/main/java/org/opennms/smoketest/containers/MockCloudContainer.java]: Done. Thanks!

**Review by fooker — APPROVED** [review]

**Pull request merged**